---
title: "R Notebook"
output: 
---

```{r}
library(data.table)
library(GenomicRanges)
library(GenomicFeatures)
```

```{r}

SVPE = fread('data/SpliceVaultPE_gtexv2.tsv.gz')

SVPE_filt = SVPE[PE_length < 450 & unann_PE_rank == 1 & prop_samples_ann > 0.1, 
                 .(chrom = gsub('chr', '', chrom), start = PE_start - 10, stop = PE_end + 10, border_match_intron, strand)]
setorder(SVPE_filt, chrom, start)

fwrite(SVPE_filt, 'data/filtered_pseudos.bed', col.names = F, sep = '\t')

```

```{r}
variant_vcf[`#CHROM` == 'Y' & POS >= 6248225 & POS <= 6248399]
```


```{r}
# get vcf of variants using precomputed scores
variant_vcf = fread('data/filtered_pseudos_sai_raw.vcf')
variant_vcf[, INFO := '.']

variant_vcf_list = split(variant_vcf, by = c('#CHROM'))
header = "##fileformat=VCFv4.2
##fileDate=20191004
##reference=GRCh38/hg38
##contig=<ID=1,length=248956422>
##contig=<ID=2,length=242193529>
##contig=<ID=3,length=198295559>
##contig=<ID=4,length=190214555>
##contig=<ID=5,length=181538259>
##contig=<ID=6,length=170805979>
##contig=<ID=7,length=159345973>
##contig=<ID=8,length=145138636>
##contig=<ID=9,length=138394717>
##contig=<ID=10,length=133797422>
##contig=<ID=11,length=135086622>
##contig=<ID=12,length=133275309>
##contig=<ID=13,length=114364328>
##contig=<ID=14,length=107043718>
##contig=<ID=15,length=101991189>
##contig=<ID=16,length=90338345>
##contig=<ID=17,length=83257441>
##contig=<ID=18,length=80373285>
##contig=<ID=19,length=58617616>
##contig=<ID=20,length=64444167>
##contig=<ID=21,length=46709983>
##contig=<ID=22,length=50818468>
##contig=<ID=X,length=156040895>
##contig=<ID=Y,length=57227415>"

lapply(seq_along(variant_vcf_list), function(x) {
  filename = paste0('data/sai_input/sai_input', x, '.vcf')
  fileConn<-file(filename)
  writeLines(header, fileConn)
  close(fileConn)
  fwrite(variant_vcf_list[[x]], file = filename, col.names = TRUE, append = TRUE, sep = '\t')})
```

```{r}
variant_vcf_list[[21]]
```




